expansion/src/main/java/no/unit/nva/expansion/model/ExpandedResource.java

```diff
@@ -189,41 +184,52 @@ private static JsonNode addFields(String json, Publication publication) {
         var sortedJson = strToJsonWithSortedContributors(json);
         injectHasFileEnum(publication, (ObjectNode) sortedJson);
         expandLicenses(sortedJson);
-        injectAnthologyRelation(publication, (ObjectNode) sortedJson);
+        injectJoinField(publication, (ObjectNode) sortedJson);
         return sortedJson;
     }
 
-    private static void injectAnthologyRelation(Publication publication, ObjectNode sortedJson) {
+    /**
+     * Injects a join field into the JSON document, intended to be used by OpenSearch. The join
+     * field is used to create a relationship between parent and child documents, where all
+     * documents are considered to either be a potential parent or a child depending on type.
+     * "Children" containing a reference to a parent will have this reference in the join field.
+     */
+    private static void injectJoinField(Publication publication, ObjectNode sortedJson) {
         Optional.ofNullable(publication.getEntityDescription())
                 .map(EntityDescription::getReference)
-                .ifPresent(reference -> addJoinFieldWhenAnthology(sortedJson, reference));
+                .ifPresent(reference -> addJoinField(sortedJson, reference));
     }
 
-    private static void addJoinFieldWhenAnthology(ObjectNode sortedJson, Reference reference) {
-        var instanceType = reference.getPublicationInstance();
+    private static void addJoinField(ObjectNode sortedJson, Reference reference) {
         var publicationContext = reference.getPublicationContext();
-
-        if (instanceType instanceof BookAnthology) {
+        var canBeParent = canBeParent(publicationContext);
+        if (canBeParent) {
             addJoinField(sortedJson, JOIN_FIELD_PARENT_LABEL, null);
-        } else if (isPartOfAnthology(publicationContext, instanceType)) {
-            var parentId = ((Anthology) publicationContext).getId();
-            if (nonNull(parentId)) {
-                var parentIdentifier = SortableIdentifier.fromUri(parentId).toString();
-                addJoinField(sortedJson, JOIN_FIELD_CHILD_LABEL, parentIdentifier);
-            } else {
-                addJoinField(sortedJson, JOIN_FIELD_CHILD_LABEL, JOIN_FIELD_DUMMY_PARENT_ID);
-            }
+        } else {
+            var parentIdentifier = getParentIdentifier(sortedJson);
+            addJoinField(sortedJson, JOIN_FIELD_CHILD_LABEL, parentIdentifier);
         }
     }
```

Comment:
Nice, but i think we have to be careful here, because for example when publicationContext is Journal, publicationContext.id is pointing to publication channel, see publication:

https://dev.nva.sikt.no/registration/019054f93468-bfe46458-f0c2-4177-ae47-22aeaf2fd727

```
"publicationContext" : {
        "type" : "Journal",
        "id" : "https://api.dev.nva.aws.unit.no/publication-channels-v2/journal/697AD9B8-B974-4F2D-8187-7B235814B622/2014"
      }
```

I think we want to have a similar check for childs as you implemented for parents, something like:

```
if (canBeChild(reference.getPublicationInstance())) {
            var parentIdentifier = getParentIdentifier(sortedJson);
            addJoinField(sortedJson, JOIN_FIELD_CHILD_LABEL, parentIdentifier);
        }
```

Not sure what should be placed in canBeChild method, but this is a good start:

```
    private static boolean canBeChild(PublicationInstance<? extends Pages> publicationInstance) {
        return publicationInstance instanceof ChapterArticle;
    }
```

Reply:
Does it matter if we set the tag for journals too? This just circles back to what we discussed yesterday, there's no obvious way to know which implementations are relevant here.

Reply:
I don't think this method will be able to extract journal identifiers:
```
private static String getPublicationIdentifier(URI uri) {
    return attempt(() -> SortableIdentifier.fromUri(uri))
            .map(SortableIdentifier::toString)
            .orElse(failure -> null);
}
```
because journal ids have year as the last path element, example: `https://api.dev.nva.aws.unit.no/publication-channels-v2/journal/C8BDE22A-49B3-45DE-8FF6-98BBFC7AA66C/2008`

Reply:
True, but then it default to the dummy value. Not ideal, but not necessarily a problem?